rust/agama-server/src/storage/web/zfcp.rs

```diff
@@ -0,0 +1,168 @@
+//! This module implements the web API for the handling of ZFCP storage service.
+//!
+//! The module offers two public functions:
+//!
+//! * `dasd_service` which returns the Axum service.
+//! * `dasd_stream` which offers an stream that emits the DASD-related events coming from D-Bus.
+
+use agama_lib::{
+    error::ServiceError, storage::{client::zfcp::ZFCPClient, model::zfcp::ZFCPController},
+};
+use axum::{
+    extract::{Path, State},
+    routing::{get, post},
+    Json, Router,
+};
+
+use crate::{error::Error, web::common::EventStreams};
+
+/// Returns the stream of ZFCP-related events.
+///
+/// The stream combines the following events:
+///
+/// * Changes on the ZFCP devices collection.
+///
+/// * `dbus`: D-Bus connection to use.
+pub async fn zfcp_stream(dbus: &zbus::Connection) -> Result<EventStreams, Error> {
+    let stream: EventStreams = vec![
+        // TODO:
+    ];
+    Ok(stream)
+}
+
+#[derive(Clone)]
+struct ZFCPState<'a> {
+    client: ZFCPClient<'a>,
+}
+
+pub async fn dasd_service<T>(dbus: &zbus::Connection) -> Result<Router<T>, ServiceError> {
+    let client = ZFCPClient::new(dbus.clone()).await?;
+    let state = ZFCPState { client };
+    let router = Router::new()
+        .route("/supported", get(supported))
```

Comment:
np: I wonder whether we should have a single end-point under `/api/storage` that reports all the supported technologies.

Reply:
maybe something like list of supported flags